numpy/__init__.pyi

```diff
@@ -1051,19 +1051,23 @@ class dtype(Generic[_DTypeScalar_co]):
     @overload
     def __getitem__(self: dtype[void], key: List[str]) -> dtype[void]: ...
     @overload
-    def __getitem__(self: dtype[void], key: Union[str, int]) -> dtype[Any]: ...
+    def __getitem__(self: dtype[void], key: str | int) -> dtype[Any]: ...
 
-    # NOTE: In the future 1-based multiplications will also yield `void` dtypes
-    @overload
-    def __mul__(self, value: L[0]) -> None: ...  # type: ignore[misc]
+    # NOTE: In the future 1-based multiplications will also yield `flexible` dtypes
     @overload
     def __mul__(self: _DType, value: L[1]) -> _DType: ...
     @overload
-    def __mul__(self, value: int) -> dtype[void]: ...
+    def __mul__(self: _FlexDType, value: SupportsIndex) -> _FlexDType: ...
+    @overload
+    def __mul__(self, value: SupportsIndex) -> dtype[void]: ...
```

Comment:
Multiplying a fixed-size dtype with 1 currently issues a `FutureWarning` and returns the passed dtype unaltered.
Out of curiosity, how long has this warning been in place and when is the switch to this new behavior planned?

``` python
In [1]: import numpy as np

In [2]: dtype_i8 = np.dtype(np.int64)
   ...: dtype_i8 * 1
<ipython-input-2-b3084159de63>:2: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  dtype_i8 * 1
Out[2]: dtype('int64')
```